addons/stock_account/models/stock_move.py

```diff
@@ -66,6 +66,37 @@ def _get_valued_types(self):
         """
         return ['in', 'out', 'dropshipped', 'dropshipped_returned']
 
+    def _filter_in_out(self):
+        in_moves, out_moves = self._get_in_out_moves()
+        is_move_in_ids = set()
+        is_move_out_ids = set()
+        for record in self:
+            if record.id in in_moves and not record._is_dropshipped_returned():
+                is_move_in_ids.add(record.id)
+
+            if record.id in out_moves and not record._is_dropshipped():
+                is_move_out_ids.add(record.id)
+
+        return is_move_in_ids, is_move_out_ids
+
+    def _get_in_out_moves(self):
```

Comment:
Except if there's a plan to extend this method I think that as the code is right now you can merge `filter_in_out` and `_get_in_out_moves` into a single function.